src/util/logging.py

```diff
@@ -17,30 +17,47 @@
 
 from util.context import Context
 
+LOG_THROTTLING_LIMIT_PER_CALLER = 10
+
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text):
+
+    # display logs called from one spot no more than the specified amount of times
+    if log_call_count.get(caller, 0) < LOG_THROTTLING_LIMIT_PER_CALLER:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
+    elif log_call_count.get(caller, 0) == LOG_THROTTLING_LIMIT_PER_CALLER:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
+        print(f"Logging calls from caller '{caller}' exceeded the throttling limit of",
+              f"{LOG_THROTTLING_LIMIT_PER_CALLER}. Further logs from this caller will be discarded")
+        return
```

Comment:
This is a bit verbose would be a bit unreadable if e.g. I entered here just to see how messages are logged, I would have to read more into it than I would like to. I can give you an example of what I think might be better.

Btw of missing things because it's too verbose - is there a print missing in the first if, or did I just get lost?